firestore/apiv1/firestore_client.go

```diff
@@ -890,7 +902,7 @@ func (c *gRPCClient) GetDocument(ctx context.Context, req *firestorepb.GetDocume
 }
 
 func (c *gRPCClient) ListDocuments(ctx context.Context, req *firestorepb.ListDocumentsRequest, opts ...gax.CallOption) *DocumentIterator {
-	md := metadata.Pairs("x-goog-request-params", fmt.Sprintf("%s=%v&%s=%v", "parent", url.QueryEscape(req.GetParent()), "collection_id", url.QueryEscape(req.GetCollectionId())))
+	md := metadata.Pairs(ReqParamsHeader, ReqParamsHeaderVal(req.GetParent()))
```

Comment:
is this needed here? This is for `collection_id`, is that getting added in elsewhere?

Reply:
Not needed. 
Looks like even the existing metadata header isn't getting used as I tested sending the request without the collection id line and it passed.
But will leave it intact to prevent regression.

Reply:
Yes.
Without this integration test fails:

```
=== RUN   TestIntegration_MissingDocs
    integration_test.go:1725: rpc error: code = InvalidArgument desc = Invalid request routing header parent=projects%2Fcndb-sdk-golang-firestore%2Fdatabases%2Fdatabase-04%2Fdocuments&collection_id=go-integration-test-20230720-78736501182000-0016. Please fill in the request header with format x-goog-request-params:project_id=cndb-sdk-golang-firestore&database_id=database-04
        error details: name = DebugInfo detail = [ORIGINAL ERROR] generic::invalid_argument: Invalid request routing header parent=projects%2Fcndb-sdk-golang-firestore%2Fdatabases%2Fdatabase-04%2Fdocuments&collection_id=go-integration-test-20230720-78736501182000-0016. Please fill in the request header with format x-goog-request-params:project_id=cndb-sdk-golang-firestore&database_id=database-04 [google.rpc.error_details_ext] { message: "Invalid request routing header parent=projects%2Fcndb-sdk-golang-firestore%2Fdatabases%2Fdatabase-04%2Fdocuments&collection_id=go-integration-test-20230720-78736501182000-0016. Please fill in the request header with format x-goog-request-params:project_id=cndb-sdk-golang-firestore&database_id=database-04" } stack =
--- FAIL: TestIntegration_MissingDocs (0.49s)
```
collection_id is not getting added elsewhere

ListDocuments gets called only in newDocumentRefIterator( https://github.com/googleapis/google-cloud-go/blob/main/firestore/list_documents.go#L58)

newDocumentRefIterator is only called at these 2 locations:
https://github.com/googleapis/google-cloud-go/blob/firestore-multi-dbs/firestore/collref.go#L130
https://github.com/googleapis/google-cloud-go/blob/firestore-multi-dbs/firestore/transaction.go#L255

I have tested both the above functions manually and in integration tests. They both pass but will recheck

Reply:
Yes.
Without this integration test fails:

```
=== RUN   TestIntegration_MissingDocs
    integration_test.go:1725: rpc error: code = InvalidArgument desc = Invalid request routing header parent=projects%2Fcndb-sdk-golang-firestore%2Fdatabases%2Fdatabase-04%2Fdocuments&collection_id=go-integration-test-20230720-78736501182000-0016. Please fill in the request header with format x-goog-request-params:project_id=cndb-sdk-golang-firestore&database_id=database-04
        error details: name = DebugInfo detail = [ORIGINAL ERROR] generic::invalid_argument: Invalid request routing header parent=projects%2Fcndb-sdk-golang-firestore%2Fdatabases%2Fdatabase-04%2Fdocuments&collection_id=go-integration-test-20230720-78736501182000-0016. Please fill in the request header with format x-goog-request-params:project_id=cndb-sdk-golang-firestore&database_id=database-04 [google.rpc.error_details_ext] { message: "Invalid request routing header parent=projects%2Fcndb-sdk-golang-firestore%2Fdatabases%2Fdatabase-04%2Fdocuments&collection_id=go-integration-test-20230720-78736501182000-0016. Please fill in the request header with format x-goog-request-params:project_id=cndb-sdk-golang-firestore&database_id=database-04" } stack =
--- FAIL: TestIntegration_MissingDocs (0.49s)
```
collection_id is not getting added elsewhere

ListDocuments gets called only in newDocumentRefIterator( https://github.com/googleapis/google-cloud-go/blob/main/firestore/list_documents.go#L58)

newDocumentRefIterator is only called at these 2 locations:
https://github.com/googleapis/google-cloud-go/blob/firestore-multi-dbs/firestore/collref.go#L130
https://github.com/googleapis/google-cloud-go/blob/firestore-multi-dbs/firestore/transaction.go#L255

I have tested both the above functions manually and in integration tests. They both pass but will recheck.

Reply:
collectionID can be passed but will be ignored by the Firestore server. Not adding collection_id